go/adbc/driver/snowflake/driver.go

```diff
@@ -162,7 +162,11 @@ func errToAdbcErr(code adbc.Status, err error) error {
 	if errors.As(err, &sferr) {
 		var sqlstate [5]byte
 		if len(sferr.SQLState) > 0 {
-			copy(sqlstate[:], sferr.SQLState[:5])
+			if len(sferr.SQLState) <= 5 {
+				copy(sqlstate[:], sferr.SQLState)
+			} else {
+				copy(sqlstate[:], sferr.SQLState[:5])
```

Comment:
Is there any penalty to always just slicing?

Should we embed SQLState into the error message always/if it's longer than 5?

Reply:
I found this because if `len(sferr.SQLState) < 5` the slicing will actually panic. 

Currently and we can't just embed the sqlstate if the error message is longer than 5 because the `AdbcError` struct defines `sqlstate` to be a `char[5]` in the header file. That said, now that I think about it, the semantics of `copy` should only copy as much as the destination will allow, so we can probably do this without the slicing here and `copy` will automatically truncate for us since the destination is a `[5]byte`. I'll update this.

Reply:
Into the error message, not into the sqlstate field.

Reply:
Ah gotcha. So SQLState will get added to the string representation of the error right now, so i think letting it truncate at 5 characters is fine with the copy.

Reply:
OK, sounds good.